api/rpc/account/account.go

```diff
@@ -0,0 +1,357 @@
+package account
+
+import (
+	"fmt"
+	"net/mail"
+
+	context "golang.org/x/net/context"
+
+	pb "github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"gopkg.in/hlandau/passlib.v1"
+)
+
+const codeLength = 8
+const hash = "$s2$16384$8$1$42JtddBgSqrJMwc3YuTNW+R+$ISfEF3jkvYQYk4AK/UFAxdqnmNFVeUw2gUVXEMBDAng=" // password
+
+// Validate validates SignUpRequest
+func (r *SignUpRequest) Validate() (err error) {
+	if r.Name == "" {
+		return grpc.Errorf(codes.InvalidArgument, "user name is mandatory")
+	}
+	address, err := mail.ParseAddress(r.Email)
+	if err != nil {
+		return grpc.Errorf(codes.InvalidArgument, err.Error())
+	}
+	r.Email = address.Address
+	if len(r.Password) < 8 {
```

Comment:
Should be a const, even though we'll want to be able to supply configurable (and stronger) password policies in the future.